code/go/0chain.net/smartcontract/dbs/event/blobber_aggregate.go

```diff
@@ -141,28 +147,27 @@ func (edb *EventDb) calculateBlobberAggregate(gs *globalSnapshot, round, limit,
 		//aggregate.TotalServiceCharge = current.TotalServiceCharge
 		aggregates = append(aggregates, aggregate)
 
-		gs.totalWritePricePeriod += aggregate.WritePrice
-
-		gs.SuccessfulChallenges += int64(aggregate.ChallengesPassed - old.ChallengesPassed)
-		gs.TotalChallenges += int64(aggregate.ChallengesCompleted - old.ChallengesCompleted)
-		gs.AllocatedStorage += aggregate.Allocated - old.Allocated
-		gs.MaxCapacityStorage += aggregate.Capacity - old.Capacity
-		gs.UsedStorage += aggregate.SavedData - old.SavedData
-		gs.TotalRewards += int64(aggregate.TotalRewards - old.TotalRewards)
+		gsDiff.SuccessfulChallenges += int64(aggregate.ChallengesPassed - old.ChallengesPassed)
+		gsDiff.TotalChallenges += int64(aggregate.ChallengesCompleted - old.ChallengesCompleted)
+		gsDiff.AllocatedStorage += aggregate.Allocated - old.Allocated
+		gsDiff.MaxCapacityStorage += aggregate.Capacity - old.Capacity
+		gsDiff.UsedStorage += aggregate.SavedData - old.SavedData
+		gsDiff.TotalRewards += int64(aggregate.TotalRewards - old.TotalRewards)
+		gsDiff.AverageWritePrice += int64(aggregate.WritePrice - old.WritePrice)
 
 		const GB = currency.Coin(1024 * 1024 * 1024)
 		if aggregate.WritePrice == 0 {
-			gs.StakedStorage = gs.MaxCapacityStorage
+			gs.StakedStorage = gs.MaxCapacityStorage	// chosed not to use gsDiff here not to over-complicate the calculation
```

Comment:
Seems `gs.ApplyDiff(&gsDiff, spenum.Blobber)` can make the `gs.StakedStorage` overflow the `gs.MaxCapacityStorage`.

Suppose we have one aggregate.WritePrice is 0, that will set the `gs.StakedStorage` to max capacity. Then in another aggregate, its write price is normal, so we will have `gsDiff.StakedStorage += ss`. In `gs.ApplyDiff(&gsDiff...)` in line 169, if we don't do the max capacity checking, the `gs.StakedStorage` will get overflow.

Reply:
Nice catch. Thanks Peter. Fixed.